### PR TITLE
chore: optimize status handshake between nodes

### PR DESCRIPTION
**Pull request by mfrankovi**
## Purpose

Refactor code around sending/receiving initial status message and making sure that nothing is sent or ignored prior to that

## Version Notes ##
{ major | minor | patch }

**NOTE: Include the version notes in the squash commit message**

### CODE REVIEW

File: src/network/network.cpp [hunk 1]
```diff
@@ -174,6 +174,14 @@ void Network::sendTransactions(NodeID const &_id, std::vector<taraxa::bytes> con
   LOG(log_dg_) << "Sent transactions:" << transactions.size();
 }
 
+void Network::setPendingPeersToReady() {
+  auto peerIds = taraxa_capability_->getAllPendingPeers();
+  for (const auto &peerId : peerIds) {
+    auto peer = taraxa_capability_->getPendingPeer(peerId);
```
Comment (MatusKysel) [inline src/network/network.cpp:180]: missing check for nullptr

File: src/network/taraxa_capability.cpp [hunk 2]
```diff
@@ -66,19 +66,38 @@ std::shared_ptr<TaraxaPeer> TaraxaCapability::getPeer(NodeID const &node_id) {
...
 
 std::shared_ptr<TaraxaPeer> TaraxaCapability::insertPeer(NodeID const &node_id) {
   boost::unique_lock<boost::shared_mutex> lock(peers_mutex_);
-  auto ret = peers_.emplace(std::make_pair(node_id, std::make_shared<TaraxaPeer>(node_id)));
+  auto ret = pending_peers_.emplace(std::make_pair(node_id, std::make_shared<TaraxaPeer>(node_id)));
+  assert(ret.second);
+  return ret.first->second;
+}
+
+std::shared_ptr<TaraxaPeer> TaraxaCapability::insertPendingPeer(NodeID const &node_id,
```
Comment (MatusKysel) [inline src/network/taraxa_capability.cpp:96]: maybe it is just me, but naming is kind of confusing
 `insertPeer` -> inserting to `pending_peers_` and `insertPendingPeer`-> inserting to `peers_` ?

Comment (justinsnapp) [reply src/network/taraxa_capability.cpp:96]: Good point @MatusKysel !   That is indeed confusing ... although I get how it happened. :-)

Comment (mfrankovi) [reply src/network/taraxa_capability.cpp]: I added new names for both methods.

File: src/network/network.cpp (hunk 1 above)
Comment (mfrankovi) [reply src/network/network.cpp:180]: Check added

**Review by MatusKysel — APPROVED** [review]

**Review by JakubFornadel — APPROVED** [review]

**Pull request merged**